contracts/src/v0.8/vrf/testhelpers/VRFV2PlusConsumerExample.sol

```diff
@@ -19,24 +21,42 @@ contract VRFV2PlusConsumerExample is ConfirmedOwner, VRFConsumerBaseV2Plus {
   }
   mapping(uint256 /* request id */ => Response /* response */) public s_requests;
 
-  constructor(
-    address vrfCoordinator,
-    address link,
-    address subOwner
-  ) ConfirmedOwner(msg.sender) VRFConsumerBaseV2Plus(vrfCoordinator, subOwner) {
+  constructor(address vrfCoordinator, address link) ConfirmedOwner(msg.sender) VRFConsumerBaseV2Plus(vrfCoordinator) {
     s_vrfCoordinator = IVRFCoordinatorV2Plus(vrfCoordinator);
     s_linkToken = LinkTokenInterface(link);
   }
 
-  function fulfillRandomWords(uint256 requestId, uint256[] memory randomWords) internal override {
+  function getRandomness(uint256 requestId, uint256 idx) public view returns (uint256 randomWord) {
     Response memory resp = s_requests[requestId];
     require(resp.requestId != 0, "request ID is incorrect");
+    return resp.randomWords[idx];
+  }
+
+  function createSubscriptionAndFund(uint96 amount) external {
```

Comment:
I wonder if it makes sense for a consumer to be it's own subscription owner? Can it still be migrated to a new coordinator in that case? I think it's still possible but it should be discouraged. This is fine for tests but we should make it clear in public facing docs that this is an anti-pattern.

Reply:
Maybe we should put a big disclaimer comment at the top here saying to not follow this pattern in production contracts.

Reply:
There are numerous examples of this, might be good for a separate PR IMO.

Reply:
Or even separate ticket.

Reply:
yeah I can add such disclaimers 👍